api/krusty/fnplugin_test.go

```diff
@@ -0,0 +1,406 @@
+package krusty_test
```

Comment:
This is a great test.
Would like to see some simpler one's in the future.
We'll need more high level fn tests in their own files (not all in one).